app/src/main/java/de/westnordost/streetcomplete/quests/access_point_ref/AddAccessPointRefForm.kt

```diff
@@ -0,0 +1,53 @@
+package de.westnordost.streetcomplete.quests.access_point_ref
+
+import android.os.Bundle
+import android.view.View
+import androidx.appcompat.app.AlertDialog
+import androidx.core.widget.doAfterTextChanged
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.data.osm.mapdata.Node
+import de.westnordost.streetcomplete.databinding.QuestRefBinding
+import de.westnordost.streetcomplete.quests.AbstractOsmQuestForm
+import de.westnordost.streetcomplete.quests.AnswerItem
+
+class AddAccessPointRefForm : AbstractOsmQuestForm<AccessPointRefAnswer>() {
+
+    override val contentLayoutResId = R.layout.quest_ref
+    private val binding by contentViewBinding(QuestRefBinding::bind)
+
+    override val otherAnswers get() = listOfNotNull(
+        AnswerItem(R.string.quest_ref_answer_noRef) { confirmNoRef() },
+        createMarkAsAssemblyPointAnswer()
+    )
+
+    private val ref get() = binding.refInput.text?.toString().orEmpty().trim()
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        binding.refInput.doAfterTextChanged { checkIsFormComplete() }
+    }
+
+    override fun onClickOk() {
+        applyAnswer(AccessPointRef(ref))
+    }
+
+    private fun confirmNoRef() {
+        AlertDialog.Builder(requireContext())
+            .setTitle(R.string.quest_generic_confirmation_title)
+            .setPositiveButton(R.string.quest_generic_confirmation_yes) { _, _ -> applyAnswer(NoAccessPointRef) }
+            .setNegativeButton(R.string.quest_generic_confirmation_no, null)
+            .show()
+    }
+
+    override fun isFormComplete() = ref.isNotEmpty()
+
+    private fun createMarkAsAssemblyPointAnswer(): AnswerItem? {
+        val node = element as? Node ?: return null
+        if (node.tags["emergency"] == "assembly_point") return null
+
+        return AnswerItem(R.string.quest_accessPointRef_answer_assembly_point) {
+            applyAnswer(IsAssemblyPointAnswer)
```

Comment:
This should also show a confirmation dialog, one that shows a picture like can be seen here https://wiki.openstreetmap.org/wiki/Tag:emergency%3Dassembly_point

Reply:
AddMaxspeedForm which shows the living street icon

Reply:
Sorry, I've looked at the implementation in `AddMaxSpeed` and it is different than I remembered after all. What you need to do is to display a custom layout in the `AlertDialog` that asks the "are you sure..." question.

This custom layout (defined in to be created `dialog_assembly_point.xml` in the `res/layout` folder would only contain an `ImageView` that shows that assembly point feature and a `TextView` that displays the text like "bla bla... looks something like this?".

The method you are looking for is `setView(layoutResId: Int)` for `AlertDialog.Builder`. E.g.

```kotlin
    private fun confirmAssemblyPoint() {
        AlertDialog.Builder(requireContext())
            .setTitle(R.string.quest_generic_confirmation_title)
            .setView(R.layout.dialog_assembly_point)
            .setPositiveButton(R.string.quest_generic_confirmation_yes) { _, _ -> applyAnswer(IsAssemblyPointAnswer) }
            .setNegativeButton(R.string.quest_generic_confirmation_no, null)
            .show()
    }
```